zipline/assets/futures.py

```diff
@@ -13,173 +13,74 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from pandas import Timestamp, Timedelta
-from pandas.tseries.tools import normalize_date
+from pandas import Timestamp
+
+from zipline.utils.input_validation import expect_types
 
 
 class FutureChain(object):
-    """ Allows users to look up future contracts.
+    """
+    Allows users to look up future contracts.
 
     Parameters
     ----------
-    asset_finder : AssetFinder
-        An AssetFinder for future contract lookups, in particular the
-        AssetFinder of the TradingAlgorithm instance.
-    get_datetime : function
-        A function that returns the simulation datetime, in particular
-        the get_datetime method of the TradingAlgorithm instance.
     root_symbol : str
         The root symbol of a future chain.
-    as_of_date : pandas.Timestamp, optional
+    as_of_date : pandas.Timestamp
         Date at which the chain determination is rooted. I.e. the
         existing contract whose notice date is first after this date is
-        the primary contract, etc. If not provided, the current
-        simulation date is used as the as_of_date.
+        the primary contract, etc.
+    chain: list
+        List of assets that represent the chain of contracts for the given
+        root symbol at the given as_of_date.
 
     Attributes
     ----------
     root_symbol : str
         The root symbol of the future chain.
-    as_of_date
+    as_of_date: Timestamp
         The current as-of date of this future chain.
-
-    Methods
-    -------
-    as_of(dt)
-    offset(time_delta)
-
-    Raises
-    ------
-    RootSymbolNotFound
-        Raised when the FutureChain is initialized with a root symbol for which
-        a future chain could not be found.
     """
-    def __init__(self, asset_finder, get_datetime, root_symbol,
-                 as_of_date=None):
-        self.root_symbol = root_symbol
-
-        # Reference to the algo's AssetFinder for contract lookups
-        self._asset_finder = asset_finder
-        # Reference to the algo's get_datetime to know the current dt
-        self._algorithm_get_datetime = get_datetime
-
-        # If an as_of_date is provided, self._as_of_date uses that
-        # value, otherwise None. This attribute backs the as_of_date property.
-        if as_of_date:
-            self._as_of_date = normalize_date(as_of_date)
-        else:
-            self._as_of_date = None
-
-        # Attribute to cache the most up-to-date chain, and the dt when it was
-        # last updated.
-        self._current_chain = []
-        self._last_updated = None
-
-        # Get the initial chain, since self._last_updated is None.
-        self._maybe_update_current_chain()
+    @expect_types(as_of_date=Timestamp)
+    def __init__(self, root_symbol, as_of_date, chain):
+        self._root_symbol = root_symbol
+        self._as_of_date = as_of_date
+        self._chain = chain
```

Comment:
What do you think about calling this `_contracts` and exposing it with a `contracts` property?

I imagine this being particularly useful in conjunction with `data.history` and `data.current`, which can take a list of contracts as an arg, so you could do:

```
CL_history = data.history(future_chain('CL').contracts, ...)
```